lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
+            warn "TIMEOUT parsing #{file}. Skipping."
+          rescue => e
```

Comment:
Gah. I hadn't actually paid attention to the code doing the rescuing in `#run`... Wouldn't this make more sense?

```ruby
        rescue => ex
          case ex
          when *RESCUABLE_ERRORS then
            $stderr.puts("Skipping file #{file} due to exception (#{ex.class}): #{ex.message}\n#{ex.backtrace.join("\n")}")
          else
            $stderr.puts("#{ex.class} error occurred processing file #{file}: aborting.")
            raise ex
          end
```